src/software/ai/evaluation/time_to_travel.cpp

```diff
@@ -0,0 +1,70 @@
+#include <cmath>
+
+#include "software/ai/evaluation/time_to_travel.h"
+
+Duration getTimeToTravelDistance(const double distance, const double max_velocity,
+                                 const double max_acceleration,
+                                 const double initial_velocity,
+                                 const double final_velocity)
+{
+    // Bound all values to be realistic
+    double d_total = std::max(0.0, distance);
+    double v_max   = std::max(0.0, max_velocity);
+    double v_i     = std::clamp(initial_velocity, -max_velocity, max_velocity);
+    double v_f     = std::clamp(final_velocity, 0.0, max_velocity);
+    double a_max   = std::max(1e-6, max_acceleration);
+
+    // Minimum distance required to accelerate/decelerate from initial to final velocity
+    double dist_required_to_reach_v_f =
+        std::abs(std::pow(v_f, 2) - std::pow(v_i, 2)) / (2 * a_max);
+    if (dist_required_to_reach_v_f > d_total)
```

Comment:
im not sure what happens here when `a_max` is `1e-6`, wouldnt we be trying to change velocities without accelerating?

Reply:
Well we would technically be accelerating, but very slowly. So the distance required to accelerate to Vf would be very large (i.e. `dist_required_to_reach_v_f`). Which would also make sense as we're diving by a very small positive number (i.e. `1e-6`)